src/server/hset_family.cc

```diff
@@ -706,6 +699,23 @@ OpResult<uint32_t> OpSet(const OpArgs& op_args, string_view key, CmdArgList valu
   return created;
 }
 
+struct HSetReplies {
+  HSetReplies(ConnectionContext* cntx)
+      : rb{static_cast<RedisReplyBuilder*>(cntx->reply_builder())} {
+    DCHECK(dynamic_cast<RedisReplyBuilder*>(cntx->reply_builder()));
+  }
+
+  template <typename T> void Send(OpResult<T> res) {
```

Comment:
I remember you added a similar approach on `Sets` (although it was not the exact same logic). What I don't remember if you added this on another family. If you did check if there is simlar logic and extract it into a base class which we can the reuse among families with this exact same pattern of replies (if they exist in the first place).